### PR TITLE
Feature/support indirect sharing mode

### PR DESCRIPTION
**Pull request by F-Shahali**
**Reference Issues/PRs**
#11 

**What does this implement/fix? Explain your changes.**
- add option to choose sharing mode (direct / indirect)
- add /share route for indirect sharing
- loading icon changed

### CODE REVIEW

File: src/Views/Layout/ShareModal/index.tsx [hunk 1]
```diff
@@ -24,13 +25,16 @@ const ShareModal = () => {
 		return `http://${url}`;
 	};
 
+	const getShareLink = (service_title: string, url: string) => {
+		return `http://localhost:3000/share/?service=${service_title}&subject=${subject}&link=${url}`;
```
Comment (AHReccese) [inline src/Views/Layout/ShareModal/index.tsx:29]: why do we have this here?
@F-Shahali

**Review by AHReccese — CHANGES_REQUESTED** [review]

File: src/Views/Layout/ShareModal/index.tsx (hunk 1 above)
Comment (F-Shahali) [reply src/Views/Layout/ShareModal/index.tsx:29]: My bad, it was for test mode. I forgot to replace `http://localhost:3000` with front domain. It's for indirect sharing.
I'll change it.

**Review by AHReccese — CHANGES_REQUESTED** [review]
First, when I'm choosing the indirect mode in the dev environment, why the generated link is for production? Everything should be in the same environment. 
Secondly, there is no need to use the `www` CNAME tag. Use the A tag, like `https://dev.mybutton.click`

**Review by AHReccese — APPROVED** [review]
LGTM

**Pull request merged**